---
title: "07-scale-up"
author: "Zach Bengtsson"
date: "2023-04-25"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{bash hisat-build}
/home/shared/hisat2-2.2.1/hisat2-build \
-f ../data/Pver_genome_assembly_v1.0.fasta \
../output/Pver_genome_assembly_v1.0-valid.index
```

``` {bash hisat}
find /home/shared/8TB_HDD_01/pver/*gz \
| xargs basename -s _R1_001.fastq.gz | xargs -I{} \
/home/shared/hisat2-2.2.1/hisat2 \
-x ../output/Pver_genome_assembly_v1.0-valid.index \
-p 8 \
-1 /home/shared/8TB_HDD_01/pver/{}_R1_001.fastq.gz \
-2 /home/shared/8TB_HDD_01/pver/{}_R2_001.fastq.gz \
-S /home/shared/8TB_HDD_01/pver/hisat-output/{}-valid.sam
```


```{bash}
find /home/shared/8TB_HDD_01/pver/hisat-output -name "*.sam" \
-exec bash -c 'file="{}"; \
basename="${file##*/}"; \
base="${basename%-valid.sam}"; \
/home/shared/samtools-1.12/samtools view -bS "$file" | \
/home/shared/samtools-1.12/samtools sort -o "/home/shared/8TB_HDD_01/pver/samtools-output/${base}-valid_sorted.bam"' \;

```

```{bash stringtie}
find /home/shared/8TB_HDD_01/pver/samtools-output/*bam \
| xargs basename -s -valid_sorted.bam | xargs -I{} \
/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie \
-p 8 \
-G ../data/Pver_genome_assembly_v1.0-valid.gtf \
-o /home/shared/8TB_HDD_01/pver/stringtie-output/{}-valid.gtf \
/home/shared/8TB_HDD_01/pver/samtools-output/{}-valid_sorted.bam
```

```{bash gffcompare}
{r, eval=FALSE, echo=TRUE}
#| code-line-numbers: "2"
/home/shared/gffcompare-0.12.6.Linux_x86_64/gffcompare \
-r ../data/Pver_genome_assembly_v1.0-valid.gtf \
-G \
-o /home/shared/8TB_HDD_01/pver/gffcompare-output/merged \
/home/shared/8TB_HDD_01/pver/stringtie-output/*.gtf \
```

```{bash filter}
awk '$3 == "transcript" && $1 !~ /^#/ {print}' /home/shared/8TB_HDD_01/pver/gffcompare-output/merged.combined.gtf | grep 'class_code "u"' | awk '$5 - $4 > 199 {print}' > /home/shared/8TB_HDD_01/pver/filter-output/merged_lncRNA_candidates.gtf
```

```{bash bedtools}
/home/shared/bedtools2/bin/bedtools \
getfasta -fi ../data/Pver_genome_assembly_v1.0.fasta -bed /home/shared/8TB_HDD_01/pver/filter-output/merged_lncRNA_candidates.gtf -fo /home/shared/8TB_HDD_01/pver/bedtools-output/merged_lncRNA_candidates.fasta -name -split
```

```{bash}
eval "$(/opt/anaconda/anaconda3/bin/conda shell.bash hook)"
python /home/shared/CPC2_standalone-1.0.1/bin/CPC2.py -i /home/shared/8TB_HDD_01/pver/bedtools-output/merged_lncRNA_candidates.fasta -o /home/shared/8TB_HDD_01/pver/cpc2-output/cpc2_results.txt
```

```{bash}
# Filter out transcripts with coding potential
awk 'NR>1 && $9 < 0 {print $1}' /home/shared/8TB_HDD_01/pver/cpc2-output/cpc2_results.txt.txt > /home/shared/8TB_HDD_01/pver/cpc2-output/noncoding_transcripts_ids.txt
grep -Fwf /home/shared/8TB_HDD_01/pver/cpc2-output/noncoding_transcripts_ids.txt /home/shared/8TB_HDD_01/pver/bedtools-output/merged_lncRNA_candidates.fasta > ~/github/zach-lncRNA/output/merged_final_lncRNAs.gtf
```

```{bash}
head ~/github/zach-lncRNA/output/merged_final_lncRNAs.gtf
```


```{r}
# Read in the file
cpc2_results <- read.table("/home/shared/8TB_HDD_01/pver/cpc2-output/cpc2_results.txt.txt", header=TRUE, sep="\t")

# View the first few rows of the file
head(cpc2_results)
```



