---
title: "09-count-matrix"
author: "Zach Bengtsson"
date: "2023-05-11"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Creating an index...

```{bash}
/home/shared/kallisto/kallisto \
index -i ~/github/zach-lncRNA/output/kallisto-output/Pver_genome_assembly_v1.0.index \
~/github/zach-lncRNA/data/Pver_genome_assembly_v1.0.fasta
```

Running kallisto on paired data...

```{bash}
find /home/shared/8TB_HDD_01/pver/*R1_001.fastq* \
| xargs basename -s _R1_001.fastq.gz | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ~/github/zach-lncRNA/output/kallisto-output/Pver_genome_assembly_v1.0.index \
-o /home/shared/8TB_HDD_01/pver/kallisto-output/{} \
-t 20 \
/home/shared/8TB_HDD_01/pver/{}_R1_001.fastq.gz \
/home/shared/8TB_HDD_01/pver/{}_R2_001.fastq.gz \
2>&1 | tee /home/shared/8TB_HDD_01/pver/kallisto-output/{}.out
```

```{bash}
perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl \
--est_method kallisto \
--gene_trans_map none \
--out_prefix ~/github/zach-lncRNA/output/count_matrix \
--name_sample_by_basedir \
/home/shared/8TB_HDD_01/pver/kallisto-output/*/abundance.tsv
```

Filtering count matrix code for just lncRNAs...

```{bash}
# Step 1: Specify the transcript ID file (GTF format)
transcript_id_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/merged_final_lncRNAs.gtf"

# Step 2: Specify the count matrix file
count_matrix_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/count_matrix.isoform.counts.matrix"

# Step 3: Specify the output file for the filtered count matrix
filtered_count_matrix_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/filtered_count_matrix.tsv"

# Extract the middle part of the transcript IDs from the GTF file
grep -oP "(?<=transcript::)[^:]*" "$transcript_id_file" > "tmp_transcript_ids.txt"

# Store the first line (column headings) of the count matrix file
head -n 1 "$count_matrix_file" > "$filtered_count_matrix_file"

# Filter the count matrix using the extracted transcript IDs, starting from the second line
awk 'NR==FNR{a[$0];next} ($1 in a)' "tmp_transcript_ids.txt" <(tail -n +2 "$count_matrix_file") >> "$filtered_count_matrix_file"

# Remove the temporary transcript ID file
rm "tmp_transcript_ids.txt"

```

```{bash}
head /home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/filtered_count_matrix.tsv
```


Leftover code for reference...
<!-- Filtering count matrix... -->
<!-- ```{bash} -->
<!-- # Step 1: Specify the transcript ID file (GTF format) -->
<!-- transcript_id_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/merged_final_lncRNAs.gtf" -->

<!-- # Step 2: Specify the count matrix file -->
<!-- count_matrix_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/count_matrix.isoform.counts.matrix" -->

<!-- # Step 3: Specify the output file for the filtered count matrix -->
<!-- filtered_count_matrix_file="/home/shared/8TB_HDD_02/zbengt/github/zach-lncRNA/output/filtered_count_matrix.tsv" -->

<!-- # Extract the middle part of the transcript IDs from the GTF file -->
<!-- grep -oP "(?<=transcript::)[^:]*" "$transcript_id_file" > "tmp_transcript_ids.txt" -->

<!-- # Filter the count matrix using the extracted transcript IDs -->
<!-- awk 'NR==FNR{a[$0];next} ($1 in a)' "tmp_transcript_ids.txt" "$count_matrix_file" > "$filtered_count_matrix_file" -->

<!-- # Remove the temporary transcript ID file -->
<!-- rm "tmp_transcript_ids.txt" -->

<!-- ``` -->








